adapter/stackdriver/log/log.go

```diff
@@ -0,0 +1,204 @@
+			continue
+		}
+		tmpl, err := template.New(name).Parse(log.PayloadTemplate)
+		if err != nil {
+			_ = logger.Errorf("failed to evaluate template for log %s, skipping: %v", name, err)
+			continue
+		}
+		infos[name] = info{
+			labels: log.LabelNames,
+			tmpl:   tmpl,
+			req:    log.HttpMapping,
+		}
+	}
+
+	client, err := b.makeClient(context.Background(), cfg.ProjectId, helper.ToOpts(cfg)...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create stackdriver logging client: %v", err)
+	}
+	return &handler{log: client.Logger("istio-mixer").Log, client: client, now: time.Now, logger: logger}, nil
+}
+
+func (h *handler) HandleLogEntry(_ context.Context, values []*logentry.Instance) error {
+	for _, v := range values {
+		linfo, found := h.info[v.Name]
+		if !found {
+			h.logger.Warningf("got instance for unknown log '%s', skipping", v.Name)
+			continue
+		}
+
+		buf := pool.GetBuffer()
+		if err := linfo.tmpl.Execute(buf, v.Variables); err != nil {
+			// We'll just continue on with an empty payload for this entry - we could still be populating the HTTP req with valuable info, for example.
+			_ = h.logger.Errorf("failed to execute template for log '%s': %v", v.Name, err)
+		}
+		payload := buf.String()
+		pool.PutBuffer(buf)
+
+		h.log(logging.Entry{
+			Timestamp:   h.now(), // TODO: use timestamp on Instance when timestamps work
+			Severity:    logging.ParseSeverity(v.Severity),
+			LogName:     v.Name,
+			Labels:      toLabelMap(linfo.labels, v.Variables),
+			Payload:     payload,
+			HTTPRequest: toReq(linfo.req, v.Variables),
+		})
+	}
+	return nil
+}
+
+func (h *handler) Close() error {
+	return h.client.Close()
+}
+
+func toLabelMap(names []string, variables map[string]interface{}) map[string]string {
+	out := make(map[string]string, len(names))
+	for _, name := range names {
+		v := variables[name]
+		switch vt := v.(type) {
+		case string:
+			out[name] = vt
+		default:
+			out[name] = fmt.Sprintf("%v", v)
+		}
+	}
+	return out
+}
+
+func toReq(mapping *config.Params_LogInfo_HttpRequestMapping, variables map[string]interface{}) *logging.HTTPRequest {
+	req := &logging.HTTPRequest{}
+	if mapping != nil {
+		reqs := variables[mapping.RequestSize]
+		reqsize, ok := toInt(reqs)
+		if ok {
+			req.RequestSize = reqsize
+		}
+
+		resps := variables[mapping.ResponseSize]
+		respsize, ok := toInt(resps)
+		if ok {
+			req.ResponseSize = respsize
+		}
+
+		code := variables[mapping.Status]
+		status, ok := code.(int)
+		if ok {
+			req.Status = status
+		}
+
+		l := variables[mapping.Latency]
+		latency, ok := l.(time.Duration)
+		if ok {
+			req.Latency = latency
+		}
+
+		lip := variables[mapping.LocalIp]
+		localip, ok := lip.(string)
+		if ok {
+			req.LocalIP = localip
+		}
+
+		rip := variables[mapping.RemoteIp]
+		remoteip, ok := rip.(string)
+		if ok {
+			req.RemoteIP = remoteip
+		}
+	}
+	return req
+}
+
+func toInt(v interface{}) (int64, bool) {
+	// case int, int8, int16, ...: return int64(i), true
```

Comment:
do we care about adding `int8`, `int16`, ... (and maybe `uint*`) ?

Reply:
I had them originally, but in practice these ints come from expression evaluation and are always int64 as a result.